### PR TITLE
[BoyScout] - moved helpers methods

### PR DESCRIPTION
**Pull request by saikrishna321**

### CODE REVIEW

File: lib/helper.js [hunk 1]
```diff
@@ -98,12 +98,66 @@ const commandlineArgs = () => {
...
+            throw new Error(`waiting failed: timeout ${timeout}ms exceeded`);
+        sleep(intervalTime);
+    }
+};
+
+const waitForNavigation = (timeout, promises = []) => {
+    return new Promise((resolve, reject) => {
+        Promise.all(promises).then(resolve);
+        const timeoutId = setTimeout(() => reject('Timedout'), timeout);
+        timeouts.push(timeoutId);
```
Comment (NivedhaSenthil) [inline lib/helper.js]: this timeout has to be cleared on closebrowser https://github.com/getgauge/taiko/blob/ca5933931bab5b273dd558abf92fc68a8c0da9f8/lib/taiko.js#L164

**Comment by saikrishna321** [comment]
@NivedhaSenthil the failure is because of the http://todomvc.com/examples/react/#/ is down. I'm going to move the failing test to http://the-internet.herokuapp.com

File: lib/helper.js (hunk 1 above)
Comment (zabil) [reply lib/helper.js]: Having generic names like `utilities.js`, `common.js` and 'helpers.js` is an anti-pattern as they become a dump yard of functions not related to each other. 

Not in the scope of this PR but at a later point we must consider splitting files according to category or domains.

**Comment by saikrishna321** [comment]
@NivedhaSenthil can you pls review now? Azure is happy 😸

**Pull request merged**